secRMM version 3.0.0 adds 'Analytic Rules' to the Microsoft Sentinel …

anthonylamark opened the pull request:
The Microsoft Sentinel team informed Squadra Technologies that the 'Microsoft Sentinel secRMM' solution was not compliant because it was missing a 'Microsoft Sentinel Analytic Rule'.  This pull request contains the secRMM Analytic Rule that was required.

   Required items:  There are no requirements for this change.
   
   Change(s):
   - Added Analytic Rules\Removable_Storage_ONLINE.yaml
   - Removed Package\2.0.1.zip
   - Added Package\3.0.0.zip
   - Modified Data\Solution_SquadraTecnologiesSecRmmTemplateSpec.json to change version number

   Reason for Change(s):
   - Added Analytic Rules folder and 1 analytic rule

   Version Updated: Bumped the version from 2.0.1 to 3.0.0

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - No

v-atulyadav commented:
Hi @anthonylamark,
It appears that your rule template does not comply with the standard format. Please refer to the link below to update your rule accordingly and then repackage the solution to include the updated changes. Thanks

https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide

anthonylamark commented:
Hi,
I have:
1. removed package 2.0.1.zip
2. uncommitted package 2.0.0.zip
3. updated WorkbooksMetadata.json with version 3.0.0 (from 1.0.0)
4. updated Removable_Storage_ONLINE.yaml
    a. changed apiVersion to 2025-09-01 (from 2023-12-01-preview)
    b. changed techniques to relevantTechniques

anthonylamark commented:
Based on the latest comment from code owner v-atulyadav:

There are still a few places where the API version needs to be upgraded. Please refer to the screenshot below for reference. Additionally, the domain validation is failing.

The following 2 files have been modified:

In Package\mainTemplate.json:
   1. modified all occurrences of:
http://www.squadratechnologies.com/StaticContent/ProductDownload/secRMM/9.9.0.0/secRMMAzureSentinelAdministratorGuide.pdf
to:
https://www.squadratechnologies.com/StaticContent/ProductDownload/secRMM/9.11.0.0/secRMMAzureSentinelAdministratorGuide.pdf
   2. modified all occurrences of:
"apiVersion": "2021-05-01"
and
"apiVersion": "2021-03-01-preview"
to:
"apiVersion": "2022-05-09"
3. modified all occurrences (lines 227 and 531)of:
"version": "1.0.0"
to:
"version": "3.0.0"

In Data Connectors\SquadraTechnologies.SecRMM.json
   1. modified 1 occurrence of:
http://www.squadratechnologies.com/StaticContent/ProductDownload/secRMM/9.9.0.0/secRMMAzureSentinelAdministratorGuide.pdf
to:
https://www.squadratechnologies.com/StaticContent/ProductDownload/secRMM/9.11.0.0/secRMMAzureSentinelAdministratorGuide.pdf

v-atulyadav commented:
Hi @anthonylamark,
We’ve resolved all failed validations. Next, we’ll review the template and then proceed with the merge. Thanks
